src/stockwatch/entities/shares.py

```diff
@@ -93,7 +93,14 @@ class SharePortfolio:
     share_positions: tuple[SharePosition, ...]
 
     def __post_init__(self) -> None:
+        the_date: date
```

Comment:
So I think my proposal for handling the date would be a bit different. Mainly due to still supporting constructing like:
```python
port = Portfolio(portfolio_date=some_date, share_positions=some_list)
```

 I guess doing something like:
```python
if self.share_positions:
    # Warn if we input weird stuff
    if self.portfolio_date and self.portfolio_date != new_date:
        # We could raise a ValueError also?
        print("Weird input date")
        print(f"Overruling portfolio date with share_position date {new_data}")
    object.__setattr__(self, "portfolio_date", self.share_positions[0].position_date)
else if self.portfolio_date is None:
    raise ValueError("Porfolio has no positions, and no date inputted, don't know what to put in")
```